data/common.yaml

```diff
@@ -0,0 +1,57 @@
+---
+ntp::authprov: ~
+ntp::broadcastclient: false
+ntp::config_dir: ~
+ntp::config_file_mode: '0644'
+ntp::config: '/etc/ntp.conf'
+ntp::config_template: ~
+ntp::config_epp: ~
+ntp::disable_auth: false
+ntp::disable_dhclient: false
+ntp::disable_kernel: false
+ntp::disable_monitor: true
+ntp::driftfile: '/var/lib/ntp/drift'
+ntp::fudge: []
+ntp::iburst_enable: true
+ntp::interfaces_ignore: []
+ntp::interfaces: []
+ntp::keys_controlkey: ~
+ntp::keys_enable: false
+ntp::keys_file: '/etc/ntp.keys'
+ntp::keys_requestkey: ~
+ntp::keys_trusted: []
+ntp::keys: []
+ntp::leapfile: ~
+ntp::logfile: ~
+ntp::ntpsigndsocket: ~
+ntp::maxpoll: ~
+ntp::minpoll: ~
+ntp::package_ensure: present
+ntp::package_manage: true
+ntp::package_name: [ 'ntp' ]
+ntp::panic: ~
+ntp::peers: []
+ntp::preferred_servers: []
+ntp::servers:
+  - '0.pool.ntp.org'
+  - '1.pool.ntp.org'
+  - '2.pool.ntp.org'
+  - '3.pool.ntp.org'
+ntp::service_enable: true
+ntp::service_ensure: running
+ntp::service_manage: true
+ntp::service_name: ntpd
+ntp::service_provider: ~
+ntp::stepout: ~
+ntp::step_tickers_file: ~
+ntp::step_tickers_template: ~
+ntp::step_tickers_epp: ~
+ntp::tinker: ~
+ntp::tos_ceiling: 15
+ntp::tos_cohort: 0
+ntp::tos_floor: 1
+ntp::tos_minclock: 3
+ntp::tos_minsane: 1
+ntp::tos: false
+ntp::udlc_stratum: 10
+ntp::udlc: false
```

Comment:
To the options hash, I have talked to puppet users who see the params list as one of the huge advantages of puppet over hand-crafting the config file, because it is a unified syntax, order-independent, and tested. My current PoV is that having the template/epp params for folks who "know better" is essential for maximum flexibility, but the main value of a module lies in *reducing* the amount of effort people have to put in.

See also our profiles-in-modules conversation from earlier this year. I haven't forgotten this, and I will come back to it!